windows-installer/install.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import print_function
+import ctypes
+import os
+import subprocess
+import sys
+
+PYTHON_BITNESS = 32
+
+
+def main():
+    script_dir_path = os.path.dirname(os.path.realpath(__file__))
+    repo_path = os.path.dirname(script_dir_path)
+    certbot_version = subprocess.check_output([sys.executable, '-c', 'import certbot; print(certbot.__version__)'],
+                                              universal_newlines=True, cwd=repo_path).strip()
+    installer_path = os.path.join(script_dir_path, 'build', 'nsis', 'certbot-{0}-installer-{1}.exe'
+                                  .format(certbot_version, 'win_amd64' if PYTHON_BITNESS == 64 else 'win32'))
```

Comment:
Current implementation is the best deal I found. Your two propositions implies higher hidden complexity in other places as a consequence.

For b, we loose the signature of Certbot version and architecture in the installer filename, which is important user side when you download a Windows installer to install a program. Reinstate it for the release pipeline will add a lot of complexity in the pipeline itself, so in Yaml, that is harder to maintain than a python script. 

For a, grouping logic needs to respect two constraints: being callable as a script for tox, and as a module for the installer script. Importing a module in a script is a brittle process if the script and the module are not in a fully fledged python project (for instance relative imports do not work from a python script). So it would require to add the usual setuptools configuration around the `windows-installer` project, which is overkill in my opinion and add a lot of complexity again.

Here, if because of the duplicated logic we break the build, it will break the associated pipelines run nightly, and we will know it. Anyway, I do not expect the installer name pattern to change regularly.

Reply:
I thought about this during the week end. I have to admit that having a windows installer file name changing for each version implies also, on top of current work, an additional complexity for the CD process. This makes the option b) more appealing in the pro/con balance.

I will go for this option b). I keep however the platform (`win32`/`win_amd64`) since using a `win_amd64` on a pure Windows 32bits would fail with obscure error codes.

It is not ideal, but the installer version is still recognizable in the GitHub release API since the version is contained in the full URL of the artifact binary. Also, version is recognizable during installation as it is displayed by the UI throughout the install process, and properly declared in Windows, so also visible in the Windows Uninstaller UI.

In the future, I think the ideal would be to have an installer not tied to a particular version of Certbot, but that dynamically install the latest version of Certbot, like for `certbot-auto`, and like a lot of program that I use and Windows, and also propose online installer/updater programs.

Reply:
Thanks for simplifying this a bit. There are a number of tradeoffs here.

I think something like what you have now is probably fine, but just for the sake of exploring our options, I don't think we'd have to make `windows-installer` a package to import a Python script in the same directory. We do this without issue in the `tools/venv*.py` scripts when we import `_venv_common`.